cerberus-health-check-lambda/src/main/groovy/com/nike/cerberus/HealthCheckHandler.groovy

```diff
@@ -0,0 +1,205 @@
+                    status: 'healthy',
+                    healthCheckPath: healthCheckPath,
+                    healthCheckValueKey: healthCheckValueKey,
+                    expectedHealthCheckValue: expectedHealthCheckValue,
+                    region: region
+            ])
+        } catch (Throwable t) {
+            return error([
+                    environment: cerberusEnvironment,
+                    status: 'unhealthy',
+                    healthCheckPath: healthCheckPath,
+                    healthCheckValueKey: healthCheckValueKey,
+                    expectedHealthCheckValue: expectedHealthCheckValue,
+                    region: region,
+                    error: true,
+                    throwableMessage: ExceptionUtils.getMessage(t),
+                    throwableMessageStacktrace: ExceptionUtils.getStackTrace(t)
+            ])
+        }
+    }
+
+    private String authenticate(AWSKMS kmsClient,
+                                OkHttpClient client,
+                                String cerberusUrl,
+                                String accountId,
+                                String roleName,
+                                String region,
+                                int retryCount = 0) {
+
+        try {
+            MediaType JSON = MediaType.parse("application/json; charset=utf-8")
+
+            RequestBody body = RequestBody.create(JSON,
+                    new JsonBuilder([
+                            account_id: accountId,
+                            role_name: roleName,
+                            region: region
+                    ]).toString())
+
+            Request request = new Request.Builder()
+                    .url("${cerberusUrl}/v1/auth/iam-role")
+                    .post(body)
+                    .build()
+            Response response = client.newCall(request).execute()
+            def bodyString = response.body().string()
+
+            assert response.code() == 200 : "Failed to get a 200 while authenticating, code: " +
+                    "${response.code()}, body: ${bodyString}"
+
+            def authPayload = new JsonSlurper().parseText(bodyString)
+
+            String encryptedAuthData = authPayload.auth_data
+            def authResp = kmsClient.decrypt(new DecryptRequest()
+                    .withCiphertextBlob(ByteBuffer.wrap(encryptedAuthData.decodeBase64())))
+            def respString = new String(authResp.getPlaintext().array())
+            def authData = new JsonSlurper().parseText(respString)
+
+            String authToken = authData?.client_token
+            assert StringUtils.isNotBlank(authToken) : 'The auth token should not be blank'
+            return authToken
+        } catch (Throwable t) {
+            if (retryCount < AUTH_RETRY_LIMIT) {
+                sleep(AUTH_RETRY_SLEEP_IN_MILLI_SECONDS)
+                return authenticate(kmsClient, client, cerberusUrl, accountId, roleName, region, retryCount + 1)
```

Comment:
Depends how you write it.  It can be done cleaner and with the code more obvious.  I'll show you what I mean later (though I might not convince you)